test/test_jit.py

```diff
@@ -1085,6 +1085,172 @@ def fn(*args):
+        self.checkScript(fn_int, (-1234, 0, ))
+        self.checkScript(fn_int, (0, -1256, ))
+        self.checkScript(fn_int, (-167, -1256, ))
+
+        def fn_float(real: float, img: float):
+            return complex(real, img)
+
+        self.checkScript(fn_float, (0.0, 0.0, ))
+        self.checkScript(fn_float, (-1234.78, 0, ))
+        self.checkScript(fn_float, (0, 56.18, ))
+        self.checkScript(fn_float, (-1.9, -19.8, ))
+
+        def fn_bool(real: bool, img: bool):
+            return complex(real, img)
+
+        self.checkScript(fn_bool, (True, True, ))
+        self.checkScript(fn_bool, (False, False, ))
+        self.checkScript(fn_bool, (False, True, ))
+        self.checkScript(fn_bool, (True, False, ))
+
+        def fn_bool_int(real: bool, img: int):
+            return complex(real, img)
+
+        self.checkScript(fn_bool_int, (True, 0, ))
+        self.checkScript(fn_bool_int, (False, 0, ))
+        self.checkScript(fn_bool_int, (False, -1, ))
+        self.checkScript(fn_bool_int, (True, 3, ))
+
+        def fn_int_bool(real: int, img: bool):
+            return complex(real, img)
+
+        self.checkScript(fn_int_bool, (0, True, ))
+        self.checkScript(fn_int_bool, (0, False, ))
+        self.checkScript(fn_int_bool, (-3, True, ))
+        self.checkScript(fn_int_bool, (6, False, ))
+
+        def fn_bool_float(real: bool, img: float):
+            return complex(real, img)
+
+        self.checkScript(fn_bool_float, (True, 0.0, ))
+        self.checkScript(fn_bool_float, (False, 0.0, ))
+        self.checkScript(fn_bool_float, (False, -1.0, ))
+        self.checkScript(fn_bool_float, (True, 3.0, ))
+
+        def fn_float_bool(real: float, img: bool):
+            return complex(real, img)
+
+        self.checkScript(fn_float_bool, (0.0, True, ))
+        self.checkScript(fn_float_bool, (0.0, False, ))
+        self.checkScript(fn_float_bool, (-3.0, True, ))
+        self.checkScript(fn_float_bool, (6.0, False, ))
+
+        def fn_float_int(real: float, img: int):
+            return complex(real, img)
+
+        self.checkScript(fn_float_int, (0.0, 1, ))
+        self.checkScript(fn_float_int, (0.0, -1, ))
+        self.checkScript(fn_float_int, (1.8, -3, ))
+        self.checkScript(fn_float_int, (2.7, 8, ))
+
+        def fn_int_float(real: int, img: float):
+            return complex(real, img)
+
+        self.checkScript(fn_int_float, (1, 0.0, ))
+        self.checkScript(fn_int_float, (-1, 1.7, ))
+        self.checkScript(fn_int_float, (-3, 0.0, ))
+        self.checkScript(fn_int_float, (2, -8.9, ))
+
+    def test_torch_complex_fails(self):
+        # The following operations are not allowed
+        # in Torchscript
+
+        with self.assertRaisesRegex(RuntimeError, "Cannot input a tensor of dimension"):
+            @torch.jit.script
+            def fn_tensor_float(real, img: float):
+                return complex(real, img)
+            fn_tensor_float(torch.rand(1), 1.2)
```

Comment:
@gmagogsfm was there a reason why you didn't enable support for this? because no error is raised when a 1D tensor with 1 element is passed to a floating point constructor 
```
@torch.jit.script
def fn_tensor_float(real):
    return float(real)
fn_tensor_float(torch.rand(1))
```

Reply:
@anjali411 : At the time when I added this test, error was raised and hence this was not enabled. I can verify and enable this test. Is there something blocking?

Reply:
no I was curious if complex constructor support for tensor inputs was not enabled for a reason. not a blocker!

Reply:
@anjali411 : This still fails for me on master.

Here is the log:

```
Traceback (most recent call last):
  File "test/test_jit.py", line 1220, in test_torch_complex_fails
    self.checkScript(fn_tensor_float_1, (torch.rand(1), 1.2, ))
  File "/data/users/nikithamalgi/pytorch/torch/testing/_internal/jit_utils.py", line 450, in checkScript
    self.checkScript(
  File "/data/users/nikithamalgi/pytorch/torch/testing/_internal/jit_utils.py", line 481, in checkScript
    script_outputs = scripted_fn(*recording_inputs)
  File "/data/users/nikithamalgi/pytorch/torch/testing/_internal/common_utils.py", line 131, in prof_func_call
    return prof_callable(func_call, *args, **kwargs)
  File "/data/users/nikithamalgi/pytorch/torch/testing/_internal/common_utils.py", line 128, in prof_callable
    return callable(*args, **kwargs)
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
  File "<string>", line 2, in fn_tensor_float_1
def fn_tensor_float_1(real, img: float):
    return complex(real, img)
           ~~~~~~~ <--- HERE
RuntimeError: Cannot input a tensor of dimension other than 0 as a scalar argument
```

Reply:
https://github.com/pytorch/pytorch/pull/52881 fixes this issue!